dnf/base.py

```diff
@@ -1413,22 +1413,32 @@ def cond_check(pkg):
                 pkg.requires in (pkg.name for pkg in self._goal._installs) or \
                 pkg.requires in [pkg.name for pkg in trans.install]
 
-        def trans_upgrade(query, remove_query):
+        def trans_upgrade(query, remove_query, pkg):
```

Comment:
doesn't this break API?

Reply:
This is internal function of def _finalize_comps_trans(self)